File: desh/ege2026kp/25solve/25-179.py
start = 800000

count = 0
x = start
while count < 5:
  d = 2
  q = round(x**0.5)
  while d <= q:
    if x % d == 0:
      break
    d += 1
  if d <= q and (d + x//d) % 138 == 0:
    print( x, d + x//d )
    count += 1
  x += 1

File: desh/ege2026kp/25solve/25-179.pas
## uses school;
(800001..maxInt).Select( x->x.Divisors )
  .Where( divs->divs.Count > 2)  
  .Select( divs->(divs[1], divs[^2], divs[^1]))
  .Where( \(d,q,x)->(d+q) mod 138 = 0  )
  .Select( \(d,q,x)-> x + ' ' + (d+q) )
  .Take(5)
  .PrintLines;